infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationHandler.java

```diff
@@ -34,39 +32,51 @@
+                                                  @Nonnull ChannelHandlerContext ctx,
+                                                  @Nonnull IClientRouter router) {
         log.debug("Handle log replication ACK");
-        return msg.getPayload();
+        return response.getPayload().getLrEntryResponse();
     }
 
-    @ClientHandler(type = CorfuMsgType.LOG_REPLICATION_METADATA_RESPONSE)
-    private static Object handleLogReplicationMetadata(CorfuPayloadMsg<LogReplicationMetadataResponse> msg,
-                                                       ChannelHandlerContext ctx, IClientRouter r) {
+    @ResponseHandler(type = PayloadCase.LR_METADATA_RESPONSE)
+    private static Object handleLogReplicationMetadata(@Nonnull ResponseMsg response,
+                                                       @Nonnull ChannelHandlerContext ctx,
+                                                       @Nonnull IClientRouter router) {
         log.debug("Handle log replication Metadata Response");
-        return msg.getPayload();
+        return response.getPayload().getLrMetadataResponse();
     }
 
-    @ClientHandler(type = CorfuMsgType.LOG_REPLICATION_QUERY_LEADERSHIP_RESPONSE)
-    private static Object handleLogReplicationQueryLeadershipResponse(CorfuPayloadMsg<LogReplicationQueryLeaderShipResponse> msg,
-                                                                      ChannelHandlerContext ctx, IClientRouter r) {
-        log.trace("Handle log replication query leadership response msg {}", msg);
-        return msg.getPayload();
+    @ResponseHandler(type = PayloadCase.LR_LEADERSHIP_RESPONSE)
+    private static Object handleLogReplicationQueryLeadershipResponse(@Nonnull ResponseMsg response,
+                                                                      @Nonnull ChannelHandlerContext ctx,
+                                                                      @Nonnull IClientRouter router) {
+        log.trace("Handle log replication query leadership response msg {}", response);
+        return response.getPayload().getLrLeadershipResponse();
     }
 
-    @ClientHandler(type = CorfuMsgType.LOG_REPLICATION_LEADERSHIP_LOSS)
-    private static Object handleLogReplicationLeadershipLoss(CorfuPayloadMsg<Messages.LogReplicationLeadershipLoss> msg,
-                                                                      ChannelHandlerContext ctx, IClientRouter r) {
-        log.debug("Handle log replication leadership loss msg {}", msg);
-        return msg.getPayload();
+    @ResponseHandler(type = PayloadCase.LR_LEADERSHIP_LOSS)
+    private static Object handleLogReplicationLeadershipLoss(@Nonnull ResponseMsg response,
+                                                             @Nonnull ChannelHandlerContext ctx,
+                                                             @Nonnull IClientRouter router) {
+        log.debug("Handle log replication leadership loss msg {}", response);
```

Comment:
Here as well: `shortDebugString` and guard.

Reply:
Done.